cmd/fluxd/main.go

```diff
@@ -596,12 +598,42 @@ func main() {
 	}
 	checkpoint.CheckForUpdates(product, version, checkpointFlags, updateCheckLogger)
 
+	if *gitAPIKey != "" {
+		parsedGitURL, err := url.Parse(*gitURL)
+
+		if err != nil {
+			logger.Log("err", err)
+			os.Exit(1)
+		}
+
+		if parsedGitURL.Scheme != "https" {
+			logger.Log("err", "git-apikey argument can be used only with HTTPS transport")
+			os.Exit(1)
+		}
+
+		if parsedGitURL.User.Username() != "" {
+			logger.Log("err", "Set git username not in git-url, but use git-user argument instead")
```

Comment:
The `--git-user` is not used in the way you think it is used, or at least, this implementation does not take into account that it is used as the _full name_ of the user (e.g. `John Doe <john@doe.com>`). While I think GitHub does some behind-the-scene parsing to still get this information presented in their UI, the data in git will be off.

However, given that we want this feature in in a short timeframe, I will volunteer to make the required changes.